azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/impl/ConvertUtils.java

```diff
@@ -0,0 +1,187 @@
+import com.microsoft.azure.toolkits.appservice.entity.WebAppDeploymentSlotEntity;
+import com.microsoft.azure.toolkits.appservice.entity.WebAppEntity;
+import com.microsoft.azure.toolkits.appservice.model.JavaVersion;
+import com.microsoft.azure.toolkits.appservice.model.OperatingSystem;
+import com.microsoft.azure.toolkits.appservice.model.PricingTier;
+import com.microsoft.azure.toolkits.appservice.model.PublishingProfile;
+import com.microsoft.azure.toolkits.appservice.model.Runtime;
+import com.microsoft.azure.toolkits.appservice.model.WebContainer;
+import com.microsoft.azure.toolkits.appservice.utils.Utils;
+import com.microsoft.azure.tools.common.model.Region;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+
+public class ConvertUtils {
+
+    static Runtime getRuntimeFromWebApp(WebAppBase webAppBase) {
+        if (StringUtils.startsWithIgnoreCase(webAppBase.linuxFxVersion(), "docker")) {
+            return Runtime.DOCKER;
+        }
+        return webAppBase.operatingSystem() == com.azure.resourcemanager.appservice.models.OperatingSystem.WINDOWS ?
+                getRuntimeFromWindowsWebApp(webAppBase) : getRuntimeFromLinuxWebApp(webAppBase);
+    }
+
+    static Runtime getRuntimeFromLinuxWebApp(WebAppBase webAppBase) {
+        if (StringUtils.isEmpty(webAppBase.linuxFxVersion())) {
+            return null;
+        }
+        final String linuxFxVersion = webAppBase.linuxFxVersion().replace("|", " ");
+        return Runtime.getRuntimeFromLinuxFxVersion(linuxFxVersion);
+    }
+
+    static Runtime getRuntimeFromWindowsWebApp(WebAppBase webAppBase) {
+        if (webAppBase.javaVersion() == null || StringUtils.isAnyEmpty(webAppBase.javaContainer(), webAppBase.javaContainerVersion())) {
+            return null;
+        }
+        final JavaVersion javaVersion = JavaVersion.values().stream()
+                .filter(version -> StringUtils.equals(webAppBase.javaVersion().toString(), version.getValue()))
+                .findFirst().orElse(null);
+        final String javaContainer = String.join(webAppBase.javaContainer(), " ", webAppBase.javaContainerVersion());
+        final WebContainer webContainer = StringUtils.equalsIgnoreCase(webAppBase.javaContainer(), "java") ? WebContainer.JAVA_SE :
+                WebContainer.values().stream()
+                        .filter(container -> StringUtils.equals(javaContainer, container.getValue()))
+                        .findFirst().orElse(null);
+        return Runtime.getRuntime(OperatingSystem.WINDOWS, webContainer, javaVersion);
+    }
+
+    static RuntimeStack convertRuntimeToRuntimeStack(Runtime runtime) {
+        if (runtime.getOperatingSystem() != OperatingSystem.LINUX || runtime.getWebContainer() == null || runtime.getJavaVersion() == null) {
+            return null;
+        }
+        final String fixedLinuxJavaVersion = runtime.getJavaVersion().getValue().startsWith(JavaVersion.JAVA_8.getValue()) ? "8" : "11";
```

Comment:
runtime.getJavaVersion should return JavaVersion.JAVA_8 or JavaVersion.JAVA_11

Reply:
Will fix in later PR.